### PR TITLE
Imaging Barrel Calorimeter Geometry and Readout Update

### PR DESCRIPTION
**Pull request by Chao1009**
Change number of staves to 24
Add z-segmentation for fibers

### Briefly, what does this PR introduce?
Geometry and readout update for the imaging barrel calorimeter.
Resolve issue #119 

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [X] Other: geometry and readout update

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [X] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
This PR requires the implementation of waveform signals in the digitization step for the hits separation in fibers.

### Does this PR change default behavior?
No

### CODE REVIEW

**Comment by Chao1009** [comment]
Working on resolving the overlap issue.

**Comment by Chao1009** [comment]
HCal barrel is also using "CaloSides" to define the number of staves. The originally assigned number of bits for staves is only 3 which cannot hold 24 staves in this change. Adjusted the number of bits to accommodate this change.

**Comment by Chao1009** [comment]
Revert the change related to Hcal. 
Now the parameter "CaloSIdes" is split into "EcalBarrelStavesN" and "HcalBarrelStavesN".

HCal keeps 12 staves while ECal is changed to 24 staves.

File: compact/ecal_barrel_interlayers.xml [hunk 1]
```diff
@@ -171,8 +171,8 @@
       <id>system:8,module:8,layer:8,slice:8,x:32:-16,y:-16</id>
     </readout>
     <readout name="EcalBarrelScFiHits">
-      <segmentation type="NoSegmentation"/>
-      <id>system:8,module:8,layer:8,slice:8,grid:16,fiber:16</id>
+      <segmentation type="CartesianStripZ" strip_size_x="1.0*cm" identifier_x="z"/>
+      <id>system:8,module:6,layer:6,slice:6,grid:10,fiber:16,z:-12</id>
```
Comment (wdconinc) [inline compact/ecal_barrel_interlayers.xml:175]: Did you verify this still allows for sufficient capacity in module, layer, slice, and grid?

**Review by wdconinc — COMMENTED** [review]
This looks good. Just one question. Can you respond here and I will approve?

I'll also restart the pipelines.

**Comment by Chao1009** [comment]
@wdconinc 
I've adjusted the bit fields for grid/z and tested it locally.
In summary, the current bit fields allow:
module (stave):     6 bits,     capacity 64,       used 24
layer:                     6 bits,     capacity 64,       used 20
slice:                      4 bits,     capacity 16,       used 6
grid:                     10 bits,     capacity 1024,   used 15
fiber:                    16 bits,     capacity 65536, used 215
z-segmentation   14 bits,     capacity 16384, allows 16+ meters long barrel

**Review by wdconinc — APPROVED** [review]

**Comment by wdconinc** [comment]
I'm going to have to merge this into a branch with eicweb permissions.

**Pull request merged**